rust/arrow/src/array/array_struct.rs

```diff
@@ -270,27 +270,26 @@ mod tests {
 
     use std::sync::Arc;
 
-    use crate::datatypes::{DataType, Field};
     use crate::{
         array::BooleanArray, array::Float32Array, array::Float64Array, array::Int32Array,
         array::StringArray, bitmap::Bitmap,
     };
+    use crate::{
+        array::Int64Array,
+        datatypes::{DataType, Field},
+    };
     use crate::{buffer::Buffer, datatypes::ToByteSlice};
 
     #[test]
     fn test_struct_array_builder() {
-        let boolean_data = ArrayData::builder(DataType::Boolean)
-            .len(4)
-            .add_buffer(Buffer::from([false, false, true, true].to_byte_slice()))
```

Comment:
This is the line that triggered this PR. As per this code shows, `to_byte_slice` can be applied to a `bool`, which promptly converts it to `u8`. However, `bool` is not represented as `u8` in the Arrow specification, but as a single bit. This indicates a problem in the `ToByteSlice`, which should not allow `bool` to be used there. I have concrete ideas to solve this.

Reply:
it might be worth filing a ticket about the issue in `ToByteSlice` - maybe others will be interesting in working / able to work on it